os/generic_unix_base.h

```diff
@@ -25,7 +25,7 @@
 #endif
 #include <sys/param.h>
 
-#if __has_include(<sys/cdefs.h>)
+#if __has_include(<sys/cdefs.h>) && defined(__GLIBC__)
```

Comment:
Wait, I think that this is going to regress the BSDs.

```suggestion
#if __has_include(<sys/cdefs.h>) && (defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__GLIBC__))
```